scripts/india_nhm/base/data_cleaner.py

```diff
@@ -0,0 +1,212 @@
+    "Manipur": "IN-MN",
+    "Meghalaya": "IN-ML",
+    "Mizoram": "IN-MZ",
+    "Nagaland": "IN-NL",
+    "Nagaland#": "IN-NL",
+    "Odisha": "IN-OR",
+    "Punjab": "IN-PB",
+    "Rajasthan": "IN-RJ",
+    "Sikkim": "IN-SK",
+    "Tamil Nadu": "IN-TN",
+    "Tamilnadu": "IN-TN",
+    "Telengana": "IN-TG",
+    "Telangana": "IN-TG",
+    "Tripura": "IN-TR",
+    "Uttarakhand": "IN-UT",
+    "Uttar Pradesh": "IN-UP",
+    "West Bengal": "IN-WB",
+    "Andaman and Nicobar Islands": "IN-AN",
+    "Andaman & Nicobar Islands": "IN-AN",
+    "Andaman & N. Island": "IN-AN",
+    "A & N Islands": "IN-AN",
+    "Chandigarh": "IN-CH",
+    "Dadra and Nagar Haveli": "IN-DN",
+    "Dadra & Nagar Haveli": "IN-DN",
+    "Dadar Nagar Haveli": "IN-DN",
+    "Daman and Diu": "IN-DD",
+    "Daman & Diu": "IN-DD",
+    "Delhi": "IN-DL",
+    "Jammu and Kashmir": "IN-JK",
+    "Jammu & Kashmir": "IN-JK",
+    "Ladakh": "IN-LA",
+    "Lakshadweep": "IN-LD",
+    "Lakshwadeep": "IN-LD",
+    "Pondicherry": "IN-PY",
+    "Puducherry": "IN-PY",
+    "Puduchery": "IN-PY",
+    "Dadra and Nagar Haveli and Daman and Diu": "IN-DN_DD",
+    "all India": "IN",
+    "all-India": "IN",
+    "All India": "IN"
+}
+
+TMCF_ISOCODE = """Node: E:{dataset_name}->E0
+typeOf: schema:Place
+isoCode: C:{dataset_name}->isoCode
+"""
+
+TMCF_NODES = """
+Node: E:{dataset_name}->E{index}
+typeOf: dcs:StatVarObservation
+variableMeasured: dcs:indianNHM/{statvar}
+measurementMethod: dcs:NHM_HealthInformationManagementSystem
+observationAbout: C:{dataset_name}->E0
+observationDate: C:{dataset_name}->Date
+observationPeriod: "P1Y"
+value: C:{dataset_name}->indianNHM/{statvar}
+"""
+
+MCF_NODES = """Node: dcid:indianNHM/{statvar}
+description: "{description}"
+typeOf: dcs:StatisticalVariable
+populationType: schema:Person
+measuredProperty: dcs:indianNHM/{statvar}
+statType: dcs:measuredValue
+
+"""
+
+
+class NHMDataLoaderBase(object):
+    """
+    An object to clean .xls files under 'data/' folder and convert it to csv
+    
+    Attributes:
+        data_folder: folder containing all the data files
+        dataset_name: name given to the dataset
+        cols_dict: dictionary containing column names in the data files mapped to StatVars
+                    (keys contain column names and values contains StatVar names)
+    """
+
+    def __init__(self, data_folder, dataset_name, cols_dict, final_csv_path):
+        """
+        Constructor
+        """
+        self.data_folder = data_folder
+        self.dataset_name = dataset_name
+        self.cols_dict = cols_dict
+        self.final_csv_path = final_csv_path
+
+        self.raw_df = None
+
+        # Ignoring the first 3 elements (State, isoCode, Date) in the dictionary map
+        self.cols_to_extract = list(self.cols_dict.keys())[3:]
+
+    def generate_csv(self):
+        """
+        Class method to preprocess the data file for each available year, extract t
+        he columns and map the columns to schema.
```

Comment:
can like line split t\nhe happen at a word boundary? Is there a pylint tool that could automatically wrap such line?

Reply:
It is the 'yapf' package for format checks. We mention the code style to be 'google' style which allows only 80 characters per line. It automatically does this wrap for comments. I will look into the abnormal comments like this. Thanks for identifying this.